concrete/src/Database/EntityManager/Provider/AbstractPackageProvider.php

```diff
@@ -0,0 +1,54 @@
+<?php
+namespace Concrete\Core\Database\EntityManager\Provider;
+
+use Concrete\Core\Application\Application;
+use Concrete\Core\Database\EntityManager\Provider\ProviderInterface;
+use Concrete\Core\Package\Package;
+use Doctrine\ORM\Mapping\Driver\AnnotationDriver;
+
+abstract class AbstractPackageProvider implements ProviderInterface
+{
+
+    /**
+     * @var Package
+     */
+    protected $pkg;
+    protected $app;
+
+    public function __construct(Package $pkg)
+    {
+        $this->app = $pkg->getApplication();
+        $this->pkg = $pkg;
+    }
+
+    protected function getLegacyAnnotationReader()
```

Comment:
Since this abstract is meant to provide these methods to subclasses, and since there's no way to resolve type here I'd say documentation is important.